src/test/java/io/vertx/ext/asyncsql/PostgreSQLTest.java

```diff
@@ -18,64 +18,83 @@
-      .put("host", System.getProperty("db.host", "localhost"))
-      .put("postgresql", new JsonObject().put("address", address));
+      .put("host", System.getProperty("db.host", "localhost"));
 
-  @Override
-  public void setUp() throws Exception {
-    super.setUp();
-    asyncSqlClient = PostgreSQLClient.createNonShared(vertx, config);
+  @Before
+  public void init() {
+    client = PostgreSQLClient.createNonShared(vertx, config);
   }
 
-  @Override
-  public void tearDown() throws Exception {
-    CountDownLatch latch;
-    if (this.asyncSqlClient != null) {
-      latch = new CountDownLatch(1);
-      this.asyncSqlClient.close((ar) -> {
-        latch.countDown();
+  @Test
+  public void someTest(TestContext context) throws Exception {
+    Async async = context.async();
+    client.getConnection(connAr -> {
+      ensureSuccess(context, connAr);
+      conn = connAr.result();
+      conn.query("SELECT 1 AS something", resultSetAr -> {
+        ensureSuccess(context, resultSetAr);
+        ResultSet resultSet = resultSetAr.result();
+        context.assertNotNull(resultSet);
+        context.assertNotNull(resultSet.getColumnNames());
+        context.assertNotNull(resultSet.getResults());
+        context.assertEquals(new JsonArray().add(1), resultSet.getResults().get(0));
+        async.complete();
       });
-      this.awaitLatch(latch);
-    }
-
-    super.tearDown();
+    });
   }
 
   @Test
-  public void someTest() throws Exception {
-    asyncSqlClient.getConnection(onSuccess(conn -> {
-      conn.query("SELECT 1 AS something", onSuccess(resultSet -> {
-        System.out.println(resultSet.getResults());
-        assertNotNull(resultSet);
-        assertNotNull(resultSet.getColumnNames());
-        assertNotNull(resultSet.getResults());
-        assertEquals(new JsonArray().add(1), resultSet.getResults().get(0));
-        conn.close((ar) -> {
-          if (ar.succeeded()) {
-            testComplete();
-          } else {
-            fail("should be able to close the asyncSqlClient");
-          }
-        });
+  public void queryTypeTimestampWithTimezoneTest(TestContext context) throws Exception {
+    Async async = context.async();
+    client.getConnection(connAr -> {
+      ensureSuccess(context, connAr);
+      conn = connAr.result();
+      conn.execute("DROP TABLE IF EXISTS test_table", onSuccess(context, dropped -> {
+        conn.execute("CREATE TABLE IF NOT EXISTS test_table (ts timestamp with time zone)", onSuccess(context, created -> {
+          conn.execute("INSERT INTO test_table (ts) VALUES (now())", onSuccess(context, inserted -> {
+            conn.query("SELECT * FROM test_table;", onSuccess(context, timestampSelect -> {
+              context.assertNotNull(timestampSelect);
+              context.assertNotNull(timestampSelect.getResults());
+              context.assertTrue(timestampSelect.getResults().get(0).getString(0).matches("\\d{4}-\\d{2}-\\d{2}T\\d{2}:\\d{2}:\\d{2}\\.\\d{3}(Z|[+-]\\d{2}:\\d{2})"));
```

Comment:
@pmlopes this would be the check I mentioned over there: https://github.com/vert-x3/vertx-mysql-postgresql-client/commit/14bb98b0fa1a9bd994a43dbb31e275ecd8e4b354#commitcomment-16058891